DATA_PREPARATION.md

```diff
@@ -0,0 +1,115 @@
+# Data Preparation Instruction
+
+## Data format
+
+This repo uses [**Coco format**](https://cocodataset.org/#home) to store the location of the images and the annotations. However, Coco format does not have a rigid requirement about the presence of fields, and it does not cover all tasks.
+
+In [COCO_DATA_FORMAT.md](COCO_DATA_FORMAT.md) doc, we make concrete requirement of what is expected in the Coco json, by this repo/pkg for various tasks, such as image classification, object detection, matting, caption, and so on.
+
+## Data folder structure
+
+A typical folder strucure can look like:
+
+```
+standord_cars/
+    train_images/
+        1.jpg
+        2.jpg
+        3.jpg
+        ...
+    test_images/
+        1.jpg
+        2.jpg
+        3.jpg
+        ...
+
+    train_images.zip
+    test_images.zip
+    train_coco.json
+    test_coco.json
+```
+
+where images are stored under `images/` and `imagez.zip`. For hosting the data on cloud or sharing, **ZIP IS NECESSARY**. Please zip (with **UNCOMPRESSED** aka **STORE** mode) the images into **one** or **multiple** zip files. You can do it with `7zip` (set compression level to 'store') on Windows or [zip](https://superuser.com/questions/411394/zip-files-without-compression) command on Linux. (Note that make sure you do have a `train` folder in your `train.zip`.)
+
+Once zipped, you can specify `"zip_file": "train_images.zip"` in the coco json for each image (check [COCO_DATA_FORMAT.md](COCO_DATA_FORMAT.md).).
+
+### Zip principles
+
+1. **Store train, val and test splits in different zip files.** For example, when doing zero-shot evaluation, all we need is the test split. By zipping them into different zips, we won't have to download a huge zip including training or val images when we just need the test images.
+2. **Make each zip file small: <= 2GB**. It is more likely for a large file to suffer from network issue or download error, so please keep each zip small. It is ok to have multiple smaller zip files.
```

Comment:
2GB restriction might be too strict, especially for larger datasets. Azure Blob client can download a file in chunks, so 100GB-ish files should be ok if correctly implemented. I suggest making it just a recommendation, not a principle.